### DISCUSSION TITLE
Hackathon: 2024-06-25

### DISCUSSION THREAD

**Discussion by asteiker** [earthaccess-dev/earthaccess#611] [Hack days]
Reporting out on earthaccess hack days.

Please create a new comment (at the very bottom, use the text box with a button that says "comment", not "reply") to write about what you did today!

Please include links to GitHub issues or pull requests relevant to your work, and use those issues or pull requests as the "source of truth" for documenting the work; this makes future readers jobs easier :) It's OK to duplicate some information while writing a summary here!

Use the "reply" feature to have a discussion under any given comment.

**Comment by mfisher87** [comment]
@itcarroll  @danielfromearth @betolink @Sherwin-14  had a great discussion about contributor accessibility.

* More common practices, less uncommon practices. Steal from scientific Python cookiecutter template? E.g. use nox, better specification of optional deps, drop conda environment files. Move some poetry config in pyproject.toml to PEP???-standard fields instead?
* Codespaces
  * Press `.` to open
  * VSCode in the browser!
  * We need to set up a dev container with our pyproject.toml, installed dependencies, etc.
  * Can we treat a live-reloading docs-editing contributor experience as the first step towards a more comprehensive code-editing experience? Seems like it can be a good way to get started.
* Discussed Git branching workflows
  * A `develop` branch which all PRs are merged into, and has less stringent checks. Releases happen on merge to main and tag with more strict tests.
  * A `release` branch; all PRs merge to `main` with less stringent checks. Releases happen on merge to `release` and tag.
  * Create "feature branches" for each PR to facilitate a handoff to maintainers. Can be automated? How will it interact with GitHub auto-generated release notes?
* We created a PR template: https://github.com/nsidc/earthaccess/pull/613
* Let's have a release party on Friday :tada: :package: :tada:

**Reply by itcarroll** [reply]
I think you mean @itcarroll.

**Reply by mfisher87** [reply]
I did. Dang autocomplete! Fixed :)

**Comment by asteiker** [comment]
@JessicaS11 @jhkennedy and I chatted about the user experience when searching for data and requesting transformation services, related to #328 :

- Looking at Earthdata Search examples of the workflow
   -  Concluding that filtering/finding data should be distinct first step, then surfacing what services are available, then requesting those and sending out to the right plug-in
   - Icepyx will hit CMR first - so this won’t need any modifications. But once we have the plug-in in place, then we could potentially remove this. 
- Discussing https://github.com/nsidc/earthaccess/issues/328 : Do we want users to have to know about harmony vs hyp3 vs icepyx? 
   - Yes, some users will be familiar with these and only want to go through hyp3. Others should have this info abstracted. 
   - For basic transformations like Harmony, no, users shouldn’t have to know or specifically go through Harmony - they just want to ask for a cropped output. 
   - Can also think about this as engine architecture: Especially for sync operations like OPeNDAP or even Harmony for basic cropping, these could be handled as sync.
   - We can start with the methods for a given service, and then build out higher level methods to abstract the service itself as a next step.

**Comment by itcarroll** [comment]
Regarding #610, neither @mfisher87 or I are 100% confident about the difference between `__getattr__` and `__getattribute__`, but some reading suggests that `__getattribute__` is called first. I'm testing whether things break whith that switch for `EarthAccessFile`,  and will submit a PR if not!

**Reply by itcarroll** [reply]
Wow! That totally breaks it. Infinite recursion ... we were warned.
